sdk/mediaservices/Azure.Media.Analytics.Edge/README.md

```diff
@@ -0,0 +1,207 @@
+    |---|---|
+    | 1.0.0b1  | 2.0  |
+
+### Creating a graph topology and making requests
+Please visit the [Examples](#examples) for starter code
+## Key concepts
+
+### MediaGraph Topology vs MediaGraph Instance
+A _graph topology_ is a blueprint or template of a graph. It defines the parameters of the graph using placeholders as values for them. A _graph instance_ references a graph topology and specifies the parameters. This way you are able to have multiple graph instances referencing the same topology but with different values for parameters. For more information please visit [Media graph topologies and instances][doc_media_graph] 
+
+### CloudToDeviceMethod
+
+The `CloudToDeviceMethod` is part of the [azure-iot-hub SDk][iot-hub-sdk]. This method allows you to communicate one way notifications to a device in your IoT hub. In our case, we want to communicate various graph methods such as `MediaGraphTopologySetRequest` and `MediaGraphTopologyGetRequest`. To use `CloudToDeviceMethod` you need to pass in two parameters: `method_name` and `payload`. 
+
+The first parameter, `method_name`, is the name of the media graph request you are sending. Make sure to use each method's predefined `method_name` property. For example, `MediaGraphTopologySetRequest.method_name`. 
+
+The second parameter, `payload`, sends the entire serialization of the media graph request. For example, `MediaGraphTopologySetRequest.serialize()`
+
+## Examples
+
+### Creating a graph topology
+To create a graph topology you need to define parameters, sources, and sinks.
+```C# Snippet:Azure_Search_Samples_SetParameters
+// Add parameters to Topology
+private void SetParameters(MediaGraphTopologyProperties graphProperties)
+{
+    graphProperties.Parameters.Add(new MediaGraphParameterDeclaration("rtspUserName", MediaGraphParameterType.String)
+    {
+        Description = "rtsp source user name.",
+        Default = "dummyUserName"
+    });
+    graphProperties.Parameters.Add(new MediaGraphParameterDeclaration("rtspPassword", MediaGraphParameterType.SecretString)
+    {
+        Description = "rtsp source password.",
+        Default = "dummyPassword"
+    });
+    graphProperties.Parameters.Add(new MediaGraphParameterDeclaration("rtspUrl", MediaGraphParameterType.String)
+    {
+        Description = "rtsp Url"
+    });
+}
+```
+
+```C# Snippet:Azure_Search_Samples_SetSourcesSinks
+// Add sources to Topology
+private void SetSources(MediaGraphTopologyProperties graphProperties)
+{
+    graphProperties.Sources.Add(new MediaGraphRtspSource("rtspSource", new MediaGraphUnsecuredEndpoint("${rtspUrl}")
+    {
+        Credentials = new MediaGraphUsernamePasswordCredentials("${rtspUserName}")
+        {
+            Password = "${rtspPassword}"
+        }
+    })
+        );
+}
+
+// Add sinks to Topology
+private void SetSinks(MediaGraphTopologyProperties graphProperties)
+{
+    graphProperties.Sinks.Add(new MediaGraphAssetSink("assetSink", new List<MediaGraphNodeInput> {
```

Comment:
Nit: Formatting.

Maybe worth to split into multiple variables for readability